src/pages/LinkDevicePage/LinkDevicePage.tsx

```diff
@@ -1,8 +1,74 @@
-import { useSetTitle, useRegisterElements } from "../../hooks";
+import { useMemo, useState, useEffect, useCallback } from "react";
+import { get, size, cloneDeep } from "lodash";
+import { useNavigate } from "react-router-dom";
+import { useDebouncedCallback } from "use-debounce";
+import {
+  LoadingSpinner,
+  useDeskproAppClient,
+  useDeskproLatestAppContext,
+} from "@deskpro/app-sdk";
+import { setEntityService } from "../../services/deskpro";
+import { useSetTitle, useRegisterElements, useAsyncError } from "../../hooks";
+import { useSearchDevices } from "./hooks";
 import { LinkDevice } from "../../components";
 import type { FC } from "react";
+import type { Maybe, UserContext } from "../../types";
+import type { Site, Device } from "../../services/lansweeper/types";
 
 const LinkDevicePage: FC = () => {
+  const navigate = useNavigate();
+  const { client } = useDeskproAppClient();
+  const { context } = useDeskproLatestAppContext() as { context: UserContext };
+  const { asyncErrorHandler } = useAsyncError();
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [siteId, setSiteId] = useState<Maybe<Site["id"]>>(null);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+  const [selectedDevices, setSelectedDevices] = useState<Device[]>([]);
+  const { isLoading, isFetching, sites, devices } = useSearchDevices(siteId, searchQuery);
+  const dpUserId = useMemo(() => get(context, ["data", "user", "id"]), [context]);
+
+  const onChangeSearchQuery = useDebouncedCallback(setSearchQuery, 1000);
```

Comment:
Magic number should probably be a const somewhere. A 1 second debounce seems high as well but maybe that's a standard?

Reply:
Made a constant and changed it to 500